### ISSUE TITLE
(Cheevos) Achievement Badges Not Displaying

### ISSUE THREAD

**Issue by Zenjir0** [libretro/RetroArch#9366]
## Description

Achievement Badges in Retroarch 1.7.8 (Windows x64) no longer display with the Badge Picture when executed, as they did in Retroarch 1.7.7 

### Expected behavior

When you acquire an achievement, and have Achievement Badges enabled along with the new Menu Widgets, you should get a drop down from the upper left corner of your screen, with your achievement badge and title of achievement being displayed, on top/over the overlay (if overlay is in use)

![image](https://user-images.githubusercontent.com/27794756/63713101-7fd18880-c82e-11e9-8708-047723661f2d.png)

### Actual behavior

When the achievement is triggered only a popup widget appears with the name and description of the achievement in the bottom left of the screen.

![image](https://user-images.githubusercontent.com/27794756/63712785-eace8f80-c82d-11e9-9c42-40e30eed700e.png)

### Bisect Results

Occurs in latest version of Retroarch 1.7.8. I only tested on Windows x64

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.7.8

### Environment information

- OS: Windows 7 x64

**Comment by negativeExponent** [comment]
i asked this also on discord a few days ago, so i confirm this to be a problem for me as well.

**Comment by natinusala** [comment]
This commit may be the culprit https://github.com/libretro/RetroArch/commit/8db00a17affd7aa897856f4b7fca729c7e2d2dbb

**Comment by negativeExponent** [comment]
badges should work now in https://github.com/libretro/RetroArch/tree/25170f36156d7f9bef2ec129134adb2c2576adcb

**Comment by twinaphex** [comment]
Hi @negativeExponent , did you confirm that this makes it work again?

**Comment by negativeExponent** [comment]
yeah but not sure if you should still see the bottom notification though as it does not appear anymore, but badges is working.
![image](https://user-images.githubusercontent.com/54053706/63776205-7b32d000-c913-11e9-8993-0a992bce3502.png)

**Comment by twinaphex** [comment]
What would that bottom notification look like? Just text?

**Comment by natinusala** [comment]
Yeah it's the runloop OSD, the fancy notification is supposed to replace that

**Comment by negativeExponent** [comment]
there should probably no "else" there in between

![image](https://user-images.githubusercontent.com/54053706/63777473-bb934d80-c915-11e9-85ae-d7ad03f698e8.png)

**Comment by twinaphex** [comment]
Achievement Unlocked: Mono Shift should probably not be shown since that is already shown at the top of the screen, but 'Complete the Prologue' probably should.

@natinusala Can you make the necessary adjustments for this?

**Comment by natinusala** [comment]
Looking at the code here https://github.com/libretro/RetroArch/blob/25170f36156d7f9bef2ec129134adb2c2576adcb/cheevos-new/cheevos.c#L507

I don't see how both notifications can be active at the same time

**Comment by negativeExponent** [comment]
probably hide this
~~snprintf(buffer, sizeof(buffer), "Achievement Unlocked: %s", cheevo->info->title);
         runloop_msg_queue_push(buffer, 0, 2 * 60, false, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
         runloop_msg_queue_push(cheevo->info->description, 0, 3 * 60, false, NULL, ~~

and leave only this?
MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);

**Comment by negativeExponent** [comment]
remove "else" there and we should have both menu_widgets_push_achievement && runloop_msg_queue_push like it used to...

or if done the way twinaphex said it should be then something like:

```
#if defined(HAVE_MENU) && defined(HAVE_MENU_WIDGETS)
      bool widgets_ready = menu_widgets_ready();
      if (widgets_ready) {
         menu_widgets_push_achievement(cheevo->info->title, cheevo->info->badge);
         runloop_msg_queue_push(cheevo->info->description, 0, 3 * 60, false, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
      }
      else
#endif
      {
         snprintf(buffer, sizeof(buffer), "Achievement Unlocked: %s", cheevo->info->title);
         runloop_msg_queue_push(buffer, 0, 2 * 60, false, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
         runloop_msg_queue_push(cheevo->info->description, 0, 3 * 60, false, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
      }
   }
```

or something like that instead of hiding the lines like ive said above (again its not pretty looking codewise)

**Comment by natinusala** [comment]
`remove "else" there and we should have both menu_widgets_push_achievement && runloop_msg_queue_push like it used to...`

It never used to have both, it was not intended and it was a bug, there should only be one and it's either the fancy notification with the badge or the regular message

**Comment by twinaphex** [comment]
Can you guys send me a PR of what it's supposed to be like?

**Comment by natinusala** [comment]
According to me the current behavior is the correct one - only the badge notification and no message at the bottom of the screen. @negativeExponent can you confirm that the current `master` has this behavior?

**Comment by negativeExponent** [comment]
yes, as of latest master, only the badge shows up.. no notifications...

https://user-images.githubusercontent.com/54053706/63776205-7b32d000-c913-11e9-8993-0a992bce3502.png

**Comment by natinusala** [comment]
Then this is the correct behavior

Le mar. 27 août 2019 à 16:51, negativeExponent <notifications@github.com> a
écrit :

**Comment by natinusala** [comment]
I mean, description can be added to the current notification - the title can fade to reveal the description. I will see if I have time to implement such a thing

**Comment by Zenjir0** [comment]
This has since been fixed with the release of 1.7.8 v2.
